awscrt/mqtt5.py

```diff
@@ -1861,6 +1875,7 @@ def puback(error_code, qos, reason_code, reason_string, user_properties_tuples):
                                      publish_packet.message_expiry_interval_sec,
                                      publish_packet.topic_alias,
                                      publish_packet.response_topic,
+                                     publish_packet.correlation_data_bytes,
                                      publish_packet.correlation_data,
```

Comment:
debatable: possibly simpler to just pass one or the other, like:
```suggestion
                                     publish_packet.correlation_data_bytes or publish_packet.correlation_data,
```

and on the C-side, just take 1 arg like:
 z* ([str](https://docs.python.org/3/library/stdtypes.html#str), [bytes-like object](https://docs.python.org/3/glossary.html#term-bytes-like-object) or None) [Py_buffer]

Reply:
That would simplify the C side significantly

Reply:
Done